src/test/java/org/wildfly/security/sasl/gs2/Gs2Test.java

```diff
@@ -524,54 +614,44 @@ private SaslClient getSaslClient(final String[] mechanisms, final String authori
         }
     }
 
-    private class Gs2ServerCallbackHandler implements CallbackHandler {
+    //TODO I don't like the indirect tests having to pass in a callback handler to get this information
+    private static class IndirectCallbackHandler implements CallbackHandler {
         private final String bindingType;
         private final byte[] bindingData;
 
-        public Gs2ServerCallbackHandler(final String bindingType, final byte[] bindingData) {
+        private IndirectCallbackHandler(String bindingType, byte[] bindingData) {
             this.bindingType = bindingType;
             this.bindingData = bindingData;
         }
 
+        @Override
         public void handle(Callback[] callbacks) throws IOException, UnsupportedCallbackException {
             for (Callback callback : callbacks) {
                 if (callback instanceof ChannelBindingCallback) {
                     final ChannelBindingCallback channelBindingCallback = (ChannelBindingCallback) callback;
                     channelBindingCallback.setBindingType(bindingType);
                     channelBindingCallback.setBindingData(bindingData);
-                } else if (callback instanceof AuthorizeCallback) {
-                    AuthorizeCallback authorizeCallback = (AuthorizeCallback) callback;
-                    authorizeCallback.setAuthorized(authorizeCallback.getAuthorizationID().equals(authorizeCallback.getAuthenticationID()));
-                } else {
-                    throw new UnsupportedCallbackException(callback);
                 }
             }
         }
-    }
+    };
 
-    private class Gs2ClientCallbackHandler implements CallbackHandler {
-        private final String bindingType;
-        private final byte[] bindingData;
-        private final boolean passCredential;
+    private static class Gs2SaslPassCredentialClientFactory extends AbstractDelegatingSaslClientFactory {
```

Comment:
Should this class be a top-level class in the `org.wildfly.security.sasl.gs2` package? Alternatively, I could change `Gs2SaslClientFactory` to include this extra stuff